src/Backend/Modules/MediaLibrary/Domain/MediaItem/MediaItemDataGrid.php

```diff
@@ -0,0 +1,197 @@
+<?php
+
+namespace Backend\Modules\MediaLibrary\Domain\MediaItem;
+
+use Backend\Core\Engine\DataGridDB;
+use Backend\Core\Engine\DataGridFunctions as BackendDataGridFunctions;
+use Backend\Core\Engine\Model;
+use Backend\Core\Language\Language;
+use Backend\Modules\MediaLibrary\Engine\Model as BackendMediaLibraryModel;
+
+/**
+ * @TODO replace with a doctrine implementation of the data grid
+ */
+class MediaItemDataGrid extends DataGridDB
+{
+    /**
+     * MediaItemDataGrid constructor.
+     *
+     * @param string $type
+     * @param int|null $folderId
+     */
+    public function __construct(string $type, int $folderId = null)
+    {
+        $andWhere = '';
+        $parameters = [(string) $type];
+
+        if ($folderId !== null) {
+            $andWhere .= ' AND i.mediaFolderId = ?';
+            $parameters[] = (int) $folderId;
+        }
+
+        parent::__construct(
+            'SELECT i.id, i.type, i.url, i.title, i.shardingFolderName, COUNT(gi.mediaItemId) as num_connected, i.mime, UNIX_TIMESTAMP(i.createdOn) AS createdOn
+             FROM MediaItem AS i
+             LEFT OUTER JOIN MediaGroupMediaItem as gi ON gi.mediaItemId = i.id
+             WHERE i.type = ?'
+                . $andWhere
+                . ' GROUP BY i.id',
+            $parameters
+        );
+
+        // filter on folder?
+        if ($folderId !== null) {
+            // set the URL
+            $this->setURL('&folder=' . $folderId, true);
+        }
+
+        // define editActionUrl
+        $editActionUrl = Model::createURLForAction('EditMediaItem');
+
+        // set headers
+        $this->setHeaderLabels(
+            array(
+                'type' => '',
+                'url' => ucfirst(Language::lbl('Image')),
+            )
+        );
+
+        // active tab
+        $this->setActiveTab('tab' . ucfirst($type));
+
+        // hide column
+        // If we have an image, show the image
+        if ($type == Type::IMAGE) {
```

Comment:
maybe refactor this into a method getColumnsThatNeedToBeHidden so you can refactor the else away